b2sdk/raw_simulator.py

```diff
@@ -831,6 +832,28 @@ def hide_file(self, api_url, account_auth_token, bucket_id, file_name):
         self.file_id_to_bucket_id[response['fileId']] = bucket_id
         return response
 
+    def copy_file(self, api_url, account_auth_token, file_id, new_file_name):
+        bucket_id = self.file_id_to_bucket_id[file_id]
+        bucket = self._get_bucket_by_id(bucket_id)
+        file_sim = bucket.file_id_to_file[file_id]
+        copy_file_sim = copy.deepcopy(file_sim)
+        copy_file_sim.upload_timestamp = six.next(self.upload_timestamp_counter)
+        copy_file_sim.action = 'copy'
+
+        response = dict(
```

Comment:
this is extremely similar to `FileSimulator.as_upload_result`, also I'm pretty sure the file is not actually saved in the simulator by this implementation so if I'll try to `ls`, I won't see it. See what `hide_file` does, it replaces an entry in a simulator state by a new entry. Unit tests would have caught this, but none were added. Please add some soon, when you add the bucket layer, for sure.

Also I think it might be necessary to refactor `as_upload_result` into something more generic, `as_operation_result` maybe, with a mandatory operation name string argument ('copy', 'upload', 'hide')